### PR TITLE
Cross-dataset: Coordinate noise augmentation (spatial jittering) at champion configs

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Adding mild Gaussian noise to input mesh coordinates during training (σ=0.001 and σ=0.005) acts as a spatial regularizer. By perturbing mesh point positions during training only, the model is forced to generalize from slightly shifted geometries rather than memorizing specific mesh discretizations. This is a standard, well-validated augmentation in point cloud learning (PointNet, PointNet++ showed consistent improvements in spatial generalization). For CFD surrogates, spatial jitter simulates manufacturing tolerances and mesh refinement variability — physically motivated regularization.

Key distinction from existing experiments:
- Point dropout (nami #2970): drops entire points — jitter keeps all points but perturbs positions
- Gradient noise (usopp #2920, confirmed FATAL): noise in weight-update space — jitter is input-space only and far less destabilizing

## Instructions

**Step 1: Add `--coord-noise-sigma` to train.py**

```python
parser.add_argument('--coord-noise-sigma', type=float, default=0.0,
                    help='Gaussian noise std added to input coords during training only')
```

**Step 2: Apply in the training loop ONLY (not eval/test)**

Find where input coordinates/positions are loaded or constructed just before the forward pass. Add immediately before `model(...)`:

```python
if args.coord_noise_sigma > 0 and model.training:
    # coords is the (N, 3) or (B, N, 3) tensor of input mesh coordinates
    coords = coords + torch.randn_like(coords) * args.coord_noise_sigma
```

Check carefully: the noise MUST be added to the spatial input (the xyz positions fed to the model's Fourier/positional encoding), NOT to feature channels. Review where `pos`, `coords`, or `x_loc` is defined in the training forward pass.

**Step 3: Run 8 experiments across all 4 benchmarks**

Use `--wandb_group mugen-coord-jitter` for all runs.

**DrivAerML (2 runs) — NO grad-clip, NO weight-decay:**
```bash
cd target/icml2026

# DM σ=0.001
SENPAI_MAX_EPOCHS=9999 python train.py --dataset drivaerml \
  --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --coord-noise-sigma 0.001 --wandb_group mugen-coord-jitter

# DM σ=0.005
SENPAI_MAX_EPOCHS=9999 python train.py --dataset drivaerml \
  --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --coord-noise-sigma 0.005 --wandb_group mugen-coord-jitter
```

**AirfRANS (2 runs):**
```bash
# AF σ=0.001
python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 \
  --weight-decay 1e-2 --no-use-ema --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 \
  --coord-noise-sigma 0.001 --wandb_group mugen-coord-jitter

# AF σ=0.005
python train.py --dataset airfrans --airfrans-task full \
  --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 \
  --weight-decay 1e-2 --no-use-ema --enable-fourier \
  --model-layers 2 --model-hidden-dim 256 --model-heads 4 --epochs 999 \
  --coord-noise-sigma 0.005 --wandb_group mugen-coord-jitter
```

**TandemFoil (2 runs):**
```bash
# TF σ=0.001
python train.py --dataset tandemfoil \
  --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 0.5 \
  --weight-decay 1e-2 --model-slices 64 --model-layers 3 \
  --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction \
  --enable-pressure-prior-addition --epochs 999 --ema-decay 0.999 \
  --coord-noise-sigma 0.001 --wandb_group mugen-coord-jitter

# TF σ=0.005
python train.py --dataset tandemfoil \
  --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 0.5 \
  --weight-decay 1e-2 --model-slices 64 --model-layers 3 \
  --model-hidden-dim 192 --model-heads 3 \
  --enable-fourier --enable-te-coord-frame --enable-cp-panel \
  --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction \
  --enable-pressure-prior-addition --epochs 999 --ema-decay 0.999 \
  --coord-noise-sigma 0.005 --wandb_group mugen-coord-jitter
```

**TandemFoil Paper (2 runs):**
```bash
# TFP σ=0.001
python train.py --dataset tandemfoil_paper \
  --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 0.5 \
  --weight-decay 1e-2 --enable-fourier --model-layers 3 \
  --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --coord-noise-sigma 0.001 --wandb_group mugen-coord-jitter

# TFP σ=0.005
python train.py --dataset tandemfoil_paper \
  --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 0.5 \
  --weight-decay 1e-2 --enable-fourier --model-layers 3 \
  --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --coord-noise-sigma 0.005 --wandb_group mugen-coord-jitter
```

**Results to report** (for each of the 8 runs):
- Best val metric and epoch
- W&B run ID
- Whether metric beat or missed baseline
- If DM σ=0.001 shows any improvement, note whether it's better or worse than σ=0.005

## Baseline

| Dataset | Metric | Baseline | W&B run | Reproduce |
|---------|--------|----------|---------|-----------|
| DrivAerML | val_primary/surface_rel_l2_pct | **3.997%** | bht6h42t | `--dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394` |
| AirfRANS | val_primary/surface_mse | **0.000482** | pr4wsbfm | `--dataset airfrans --airfrans-task full --optimizer adamw --lr 6e-4 --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 --no-use-ema --enable-fourier --model-layers 2 --model-hidden-dim 256 --model-heads 4` |
| TandemFoil | val_primary/surface_pressure_mae | **22.537** | 0lv7fnun | `--dataset tandemfoil --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 0.5 --weight-decay 1e-2 --model-slices 64 --model-layers 3 --model-hidden-dim 192 --model-heads 3 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --ema-decay 0.999` |
| TandemFoil Paper | val_primary/field_mse | **0.002383** | d1xh0o1p | `--dataset tandemfoil_paper --optimizer lion --lr 1.25e-4 --cosine-t-max 10 --grad-clip 0.5 --weight-decay 1e-2 --enable-fourier --model-layers 3 --model-hidden-dim 192 --model-heads 3 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --ema-decay 0.999` |

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing to narrow the queue for the final ICML benchmark sprint. Current priority is benchmark-specific optimization and benchmark-faithful reporting for DrivAerML, TandemFoil paper benchmark, AirfRANS full Surf/Vol, and the Tandem parity anchor. If this idea still matters later, we can reopen it in a benchmark-specific form.
